crates/rpc/rpc/src/eth/api/transactions.rs

```diff
@@ -315,17 +315,15 @@ where
         };
 
         let len = res_receipt.logs.len() as u64;
-        res_receipt.logs.iter_mut().for_each(|log| {
-            let mut _index = 0;
+        for (idx, log) in res_receipt.logs.iter_mut().enumerate() {
             log.transaction_hash = Some(meta.tx_hash);
             log.transaction_index = Some(U256::from(meta.index));
             log.block_hash = Some(meta.block_hash);
             log.block_number = Some(U256::from(meta.block_number));
-            log.log_index = Some(U256::from(meta.index * (len - 1) + _index));
-            log.transaction_log_index = Some(U256::from(_index));
+            log.log_index = Some(U256::from(meta.index * (len - 1) + idx as u64));
```

Comment:
this is not correct, since this is supposed to be the idx of the log in the block so we number of logs prior to this tx in the block

Reply:
Oh yes this should be changed